### PR TITLE
Consensus/Core unit test fixes and documentation improvements

### LINKED ISSUE #24: Fix unit tests for Consensus contract (feature/consensus)

### PR DESCRIPTION
**Pull request by abhayks1**
PR covers below:

- Failing unit test fixes for Consensus and Core contract
- Documentation improvements
- Moved few hardcoded values to constant variable
- Renaming of variable `joinLimit` to `maxValidators`

Axiom contract test fixes are being taken care with ticket #26 

Fixes #24

### CODE REVIEW

**Comment by abhayks1** [comment]
PR is ready for review 🍏

File: contracts/test/consensus/MockConsensus.sol [hunk 1]
```diff
@@ -127,7 +127,7 @@ contract MockConsensus is ConsensusI, ReputationI {
     }
 
     function newMetaChain(
-        bytes20 _chainId,
+        address _chainId,
```
Comment (deepesh-kn) [inline contracts/test/consensus/MockConsensus.sol]:
```suggestion
        address _anchor,
```

File: test/consensus/join_during_creation.js [hunk 2]
```diff
@@ -104,49 +105,49 @@ contract('Consensus::joinDuringCreation', (accounts) => {
...
+      await consensus.setCoreStatus(core.address, CoreStatusUtils.CoreStatus.creation);
+      await consensusUtil.joinDuringCreation(consensus, joinParams);
+    });
+
+    it('should pass when core status is opened', async () => {
+      await consensus.setCoreStatus(core.address, CoreStatusUtils.CoreStatus.opened);
+      await consensusUtil.joinDuringCreation(consensus, joinParams);
+    });
+
+    it('should pass when core status is opened', async () => {
```
Comment (deepesh-kn) [inline test/consensus/join_during_creation.js]: This is a repetition of previous test.

**Review by deepesh-kn — CHANGES_REQUESTED** [review]
Nice work 👍 
Just a few minor changes.

File: contracts/test/consensus/MockConsensus.sol (hunk 1 above)
Comment (abhayks1) [reply contracts/test/consensus/MockConsensus.sol]: Done!

File: test/consensus/join_during_creation.js (hunk 2 above)
Comment (abhayks1) [reply test/consensus/join_during_creation.js]: Duplication removed.

**Review by deepesh-kn — APPROVED** [review]

**Pull request merged**